bindings/python/addtags.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Dependencies:
+# - auditwheel>=3,<4
+# Usage:
+# $ python addtags.py WHEEL_PATH TARGET_PLATFORM WHEEL_DIR
+import os
+from os.path import abspath, isfile, exists, basename, join as pjoin
+from sys import argv
+from auditwheel.policy import get_priority_by_name, get_replace_platforms
+from auditwheel.wheel_abi import analyze_wheel_abi
+from auditwheel.wheeltools import InWheelCtx, add_platforms
+
+
+def repair_wheel(wheel_path, abi, wheel_dir):
+    wheel_fname = basename(wheel_path)
+    with InWheelCtx(wheel_path) as ctx:
+        ctx.out_wheel = pjoin(wheel_dir, wheel_fname)
+        ctx.out_wheel = add_platforms(ctx, [abi],
+                                      get_replace_platforms(abi))
+    return ctx.out_wheel
+
+
+def main(wheel_path, abi, wheel_dir):
+    if not isfile(wheel_path):
+        raise FileNotFoundError('cannot access wheel file %s' % (wheel_path,))
+
+    if not exists(wheel_dir):
+        os.makedirs(wheel_dir)
+
+    reqd_tag = get_priority_by_name(abi)
+    out_wheel = repair_wheel(wheel_path, abi, wheel_dir)
+
+    if out_wheel is not None:
+        analyzed_tag = analyze_wheel_abi(out_wheel).overall_tag
+        if reqd_tag < get_priority_by_name(analyzed_tag):
+            print('Wheel is eligible for a higher priority tag. '
+                  'You requested %s but I have found this wheel is '
+                  'eligible for %s.' % (abi, analyzed_tag))
+            out_wheel = repair_wheel(wheel_path, analyzed_tag, wheel_dir)
```

Comment:
Can you explain this code?

Reply:
This happens when you build a manylinux1 wheel and attempt to assign it a manylinux2010 tag. Then auditwheel essentially tells you that you can support installation on a wider array of systems by simply using a more permissive ('higher priority') tag and then that's what it does.
See this file for a list of policies and associated priorities - https://github.com/pypa/auditwheel/blob/master/auditwheel/policy/policy.json - you will note that the most permissive policy, i.e. the policy with the lowest minimum requirements have highest priority.